src/rdkafka_int.h

```diff
@@ -339,6 +337,42 @@ struct rd_kafka_s {
                 /**< Lock for sparse_connect_random */
                 mtx_t         sparse_connect_lock;
         } rk_suppress;
+
+        /**
+         * SASL/OAUTHBEARER token value, token metadata, and extensions
+         */
+        struct {
+                /**< The lock protecting the below information */
+                rwlock_t refresh_lock;
+
+                /**< The b64token value as defined in RFC 6750 Section 2.1
+                 *   https://tools.ietf.org/html/rfc6750#section-2.1
+                 */
+                char *token_value;
+                /**< When the token expires, in terms of the number of
+                 *   milliseconds since the epoch.
+                 */
+                int64_t md_lifetime_ms;
```

Comment:
uint64_t

Reply:
Fixed.  Did not change signature of set_token() -- it still takes a signed value -- but it does check to make sure it is greater than "now", which will always be positive.